### PR TITLE
Add hazelcast-full-example.yaml

### PR DESCRIPTION
**Pull request by blazember**
The change adds `hazelcast-full-example.yaml`.
Besides adding this file, `hazelcast-full-example.xml` is slightly changed to make it a configuration that doesn't violate any checks in the config builders and adds a few missing configuration elements.

A test is added that verifies that the xml and yaml example files are resolved to the same Config structure.

Copying the yaml files to the target to be part of the built zip.

Extra notes:
Given the `hazelcast-full-example.xml` was invalid before the change and a few configuration elements, attributes were missing, we need to spend some effort on verifying, updating the example. Also, the comments describing the YAML configuration options should be better tailored to YAML. This was not done due to the time constraints.

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 1]
```diff
@@ -525,12 +527,17 @@
         </outbound-ports>
         <reuse-address>false</reuse-address>
         <join>
-            <multicast enabled="true">
+            <multicast enabled="false">
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: Needed to make the example valid.

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 2]
```diff
@@ -1133,7 +1141,7 @@
         <metadata-policy>CREATE_ON_UPDATE</metadata-policy>
         <statistics-enabled>true</statistics-enabled>
         <optimize-queries>true</optimize-queries>
-        <cache-deserialized-values>INDEX-ONLY</cache-deserialized-values>
+        <cache-deserialized-values>ALWAYS</cache-deserialized-values>
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: Needed to make the example valid.

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 3]
```diff
@@ -1474,7 +1482,7 @@
                         class-name="com.example.cache.MyEntryEventFilterFactory"/>
             </cache-entry-listener>
         </cache-entry-listeners>
-        <in-memory-format>NATIVE</in-memory-format>
+        <in-memory-format>BINARY</in-memory-format>
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: Needed to make the be able to load the config in OS without error.

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 4]
```diff
@@ -2221,14 +2229,14 @@
                 </endpoints>
             </all-permissions>
             <config-permission>
-                <actions>
-                    <action>all</action>
-                </actions>
+                <endpoints>
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: These `<action>`s were invalid here and below.

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 5]
```diff
@@ -0,0 +1,2515 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a full example hazelcast.yaml that includes all the
+# configuration elements and attributes of a Hazelcast member.
+#
+# To use this, rename it to hazelcast.xml and place it in
```
Comment (mustafaiman) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: `hazelcast.xml` -> `hazelcast.yaml`

Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.yaml]: Nice catch :+1:

**Comment by blazember** [comment]
run-lab-run

**Comment by blazember** [comment]
run-lab-run

**Comment by blazember** [comment]
Merging this PR with one unrelated test failure in the last build since the PR cherry-picked from this PR is already released with 3.12-BETA-1, and the files in the PR are example declarative configuration files and their new passing test case.

**Pull request merged**